python/sglang/srt/models/torch_native_llama.py

```diff
@@ -176,7 +165,6 @@ def __init__(
     ) -> None:
         super().__init__()
         self.hidden_size = hidden_size
-        tp_size = get_tensor_model_parallel_world_size()
         self.total_num_heads = num_heads
```

Comment:
I see it seems that we are already doing manual sharding here, I do feel this code should move to separate tp related code instead of being embedded in the model if possible

Reply:
Yeah, it seems the "local" `n_heads` are needed for constructing the `RadixAttention` later.
```
self.attn = RadixAttention(
    self.num_heads,
    ...
```
I am not sure if I can remove it given that it involves a contract change with that module.

Reply:
I think ideally we want to remove all tp logic outside of the model, that way we can apply this to other models without modifying model at all